src/ElasticApmAgentLaravelOctane/OctaneApmManager.php

```diff
@@ -151,8 +204,43 @@ public function endTransaction(): void
             throw new BadMethodCallException('Cannot start transaction before it has been started');
         }
 
-        if ( ! $this->transaction->hasEnded()) {
-            $this->transaction->end();
+        foreach (array_keys($this->spans) as $spanKey) {
```

Comment:
Is this neccessary?

Reply:
It is only used in case of errors. Without problems the array is an empty list.